app/models/account.rb

```diff
@@ -17,15 +17,24 @@ def find_or_create_accounts_resource
 
     INVALID_ID_CHARS = /[ :]/.freeze
 
+    def host_slosilo_id(account)
+      "authn:#{account}:host"
+    end
+
+    def user_slosilo_id(account)
+      "authn:#{account}:user"
+    end
+
     def create(id, owner_id = nil)
-      raise Exceptions::RecordExists.new("account", id) if Slosilo["authn:#{id}"]
+      raise Exceptions::RecordExists.new("account", id) if Slosilo[host_slosilo_id(id)] or Slosilo[user_slosilo_id(id)]
```

Comment:
Alternatively, you can use the `token_key...` methods below (which return the key if present:

```ruby
raise Exceptions::RecordExists.new("account", id) if token_key('user') || token_key('host')
```

Reply:
You likely know this, but this line will raise if either host or user keys are missing. It the migration the only bit ensuring both keys are present?

Reply:
what do you mean? there is a rspec test to test it